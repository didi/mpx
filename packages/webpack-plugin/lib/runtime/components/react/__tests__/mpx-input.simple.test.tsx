import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../mpx-input'

// Mock dependencies
jest.mock('@mpxjs/utils', () => ({
  warn: jest.fn(),
  error: jest.fn(),
  isFunction: jest.fn((fn) => typeof fn === 'function')
}))

jest.mock('../utils', () => ({
  parseUrl: jest.fn(),
  PERCENT_REGEX: /%$/,
  isIOS: false,
  useUpdateEffect: jest.fn((effect, deps) => {
    const mockReact = require('react')
    mockReact.useEffect(effect, deps)
  }),
  useTransformStyle: jest.fn((style) => ({
    normalStyle: style || {},
    hasSelfPercent: false,
    hasPositionFixed: false,
    hasVarDec: false,
    varContextRef: { current: {} },
    setWidth: jest.fn(),
    setHeight: jest.fn()
  })),
  useLayout: jest.fn(() => ({
    layoutRef: { current: null },
    layoutStyle: {},
    layoutProps: {}
  })),
  extendObject: jest.fn((...args) => Object.assign({}, ...args))
}))

jest.mock('../getInnerListeners', () => ({
  __esModule: true,
  default: jest.fn((props) => props),
  getCustomEvent: jest.fn((type, evt, ref, props) => ({
    type,
    target: {
      value: evt?.nativeEvent?.text || evt?.nativeEvent?.value || ''
    },
    detail: {
      value: evt?.nativeEvent?.text || evt?.nativeEvent?.value || '',
      cursor: 0
    }
  }))
}))

jest.mock('../useNodesRef', () => ({
  __esModule: true,
  default: jest.fn()
}))

jest.mock('../context', () => {
  const mockReact = require('react')
  return {
    FormContext: mockReact.createContext(null),
    KeyboardAvoidContext: mockReact.createContext(null)
  }
})

jest.mock('../mpx-portal', () => {
  const mockReact = require('react')
  return {
    __esModule: true,
    default: mockReact.forwardRef((props: any, ref: any) => {
      return mockReact.createElement('View', { ...props, ref })
    })
  }
})

describe('MpxInput Simple Tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  // 基础渲染测试
  it('should render with default props', () => {
    const component = renderer.create(<Input adjust-position={true} />)
    const tree = component.toJSON()
    expect(tree).toBeTruthy()
    expect(tree).toMatchSnapshot()
  })

  it('should render with value prop', () => {
    const component = renderer.create(
      <Input value="test value" adjust-position={true} />
    )
    const tree = component.toJSON()
    expect(tree).toBeTruthy()
    expect(tree).toMatchSnapshot()
  })

  it('should render with placeholder', () => {
    const component = renderer.create(
      <Input placeholder="Enter text here" adjust-position={true} />
    )
    const tree = component.toJSON()
    expect(tree).toBeTruthy()
    expect(tree).toMatchSnapshot()
  })

  // 输入类型测试
  it('should handle text type', () => {
    const component = renderer.create(
      <Input type="text" value="text input" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should handle number type', () => {
    const component = renderer.create(
      <Input type="number" value="123" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should handle password input', () => {
    const component = renderer.create(
      <Input password value="secret" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 状态测试
  it('should handle disabled state', () => {
    const component = renderer.create(
      <Input disabled value="disabled" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should handle auto-focus', () => {
    const component = renderer.create(
      <Input auto-focus value="auto focus" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 约束测试
  it('should handle maxlength', () => {
    const component = renderer.create(
      <Input maxlength={10} value="short" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 样式测试
  it('should handle custom styles', () => {
    const style = {
      fontSize: 16,
      color: '#333',
      backgroundColor: '#f5f5f5'
    }
    const component = renderer.create(
      <Input style={style} value="styled" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 事件处理测试
  it('should handle input events', () => {
    const mockOnInput = jest.fn()
    const component = renderer.create(
      <Input bindinput={mockOnInput} value="test" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should handle focus events', () => {
    const mockOnFocus = jest.fn()
    const component = renderer.create(
      <Input bindfocus={mockOnFocus} value="test" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 边界情况测试
  it('should handle empty value', () => {
    const component = renderer.create(
      <Input value="" adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should handle numeric value', () => {
    const component = renderer.create(
      <Input value={123} adjust-position={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  // 更新测试
  it('should update value correctly', () => {
    const component = renderer.create(
      <Input value="initial" adjust-position={true} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot('initial value')

    // 更新组件
    component.update(
      <Input value="updated" adjust-position={true} />
    )

    tree = component.toJSON()
    expect(tree).toMatchSnapshot('updated value')
  })

  it('should update disabled state', () => {
    const component = renderer.create(
      <Input value="test" disabled={false} adjust-position={true} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot('enabled state')

    // 更新到禁用状态
    component.update(
      <Input value="test" disabled={true} adjust-position={true} />
    )

    tree = component.toJSON()
    expect(tree).toMatchSnapshot('disabled state')
  })
})
