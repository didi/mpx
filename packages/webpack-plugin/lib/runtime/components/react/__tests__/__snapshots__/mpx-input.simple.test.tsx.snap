// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MpxInput Simple Tests should handle auto-focus 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  auto-focus={true}
  autoFocus={true}
  blurOnSubmit={true}
  defaultValue="auto focus"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="auto focus"
/>
`;

exports[`MpxInput Simple Tests should handle custom styles 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="styled"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#f5f5f5",
      "color": "#333",
      "fontSize": 16,
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="styled"
/>
`;

exports[`MpxInput Simple Tests should handle disabled state 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="disabled"
  disabled={true}
  editable={false}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="disabled"
/>
`;

exports[`MpxInput Simple Tests should handle empty value 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue=""
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value=""
/>
`;

exports[`MpxInput Simple Tests should handle focus events 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  bindfocus={[MockFunction]}
  blurOnSubmit={true}
  defaultValue="test"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="test"
/>
`;

exports[`MpxInput Simple Tests should handle input events 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  bindinput={[MockFunction]}
  blurOnSubmit={true}
  defaultValue="test"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="test"
/>
`;

exports[`MpxInput Simple Tests should handle maxlength 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="short"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={10}
  maxlength={10}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="short"
/>
`;

exports[`MpxInput Simple Tests should handle number type 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="123"
  editable={true}
  enterKeyHint="done"
  keyboardType="numeric"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  type="number"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="123"
/>
`;

exports[`MpxInput Simple Tests should handle numeric value 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="123"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="123"
/>
`;

exports[`MpxInput Simple Tests should handle password input 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="secret"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  password={true}
  secureTextEntry={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="secret"
/>
`;

exports[`MpxInput Simple Tests should handle text type 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="text input"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  type="text"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="text input"
/>
`;

exports[`MpxInput Simple Tests should render with default props 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue=""
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value=""
/>
`;

exports[`MpxInput Simple Tests should render with placeholder 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue=""
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  placeholder="Enter text here"
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value=""
/>
`;

exports[`MpxInput Simple Tests should render with value prop 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="test value"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="test value"
/>
`;

exports[`MpxInput Simple Tests should update disabled state: disabled state 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="test"
  disabled={true}
  editable={false}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="test"
/>
`;

exports[`MpxInput Simple Tests should update disabled state: enabled state 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="test"
  disabled={false}
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="test"
/>
`;

exports[`MpxInput Simple Tests should update value correctly: initial value 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="initial"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="initial"
/>
`;

exports[`MpxInput Simple Tests should update value correctly: updated value 1`] = `
<TextInput
  adjust-position={true}
  allowFontScaling={false}
  autoFocus={false}
  blurOnSubmit={true}
  defaultValue="updated"
  editable={true}
  enterKeyHint="done"
  keyboardType="default"
  maxLength={140}
  multiline={false}
  onBlur={[Function]}
  onChange={[Function]}
  onContentSizeChange={[Function]}
  onFocus={[Function]}
  onSelectionChange={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  secureTextEntry={false}
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 0,
    }
  }
  textAlignVertical="auto"
  underlineColorAndroid="rgba(0,0,0,0)"
  value="initial"
/>
`;
